parquet-column/src/main/java/org/apache/parquet/column/statistics/geometry/EnvelopeCovering.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.column.statistics.geometry;
+
+import java.nio.ByteBuffer;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.Envelope;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.GeometryFactory;
+import org.locationtech.jts.io.ParseException;
+import org.locationtech.jts.io.WKBReader;
+import org.locationtech.jts.io.WKBWriter;
+
+public class EnvelopeCovering extends Covering {
+
+  private static final ByteBuffer EMPTY = ByteBuffer.wrap(new byte[0]);
+  private final WKBReader reader = new WKBReader();
+  private final WKBWriter writer = new WKBWriter();
+  private final GeometryFactory factory = new GeometryFactory();
+
+  public EnvelopeCovering() {
+    super(EMPTY, LogicalTypeAnnotation.Edges.SPHERICAL);
+  }
+
+  @Override
+  void update(Geometry geom) {
+    if (geom == null) {
+      return;
+    }
+    try {
+      if (geometry != EMPTY) {
+        Geometry existingGeometry = reader.read(geometry.array());
+        Envelope existingEnvelope = existingGeometry.getEnvelopeInternal();
+        Envelope newEnvelope = geom.getEnvelopeInternal();
+
+        // The following expandToInclude method works correctly for planar coordinate systems.
+        // It simply extends the existing envelope to include the new envelope by adjusting
+        // the min and max values of x and y coordinates.
+        existingEnvelope.expandToInclude(newEnvelope);
+
+        // However, this approach is not accurate for spherical coordinate systems.
```

Comment:
If this algorithm does not work for spherical geometry, should we throw when the logical type has set edges attribute to spherical?

Reply:
@wgtmac Do you mean throw an exception? If so, I agree. We can throw an exception saying this implementation does not support spherical edges.

Reply:
Yes, but `GeometryStatistics.supportsCovering()` indicates that covering is built for spherical edges only, which is on the contrary of this comment.

Reply:
@wgtmac @jiayuasu Ok, i've tried to implement the envelope covering for spherical coordinates using the net.sf.geographiclib.Geodesic library. Due to the complexity, I only plan to support the "EPSG:3857" for the time. Let me know if this approach works for you.